File: artifacts/quadtree-0.1.0/translate/src/point.rs
use crate::*;

pub fn quadtree_point_new(x: f64, y: f64) -> Option<Box<QuadtreePoint>> {
    Some(Box::new(QuadtreePoint { x, y }))
}
pub fn quadtree_point_free(point: Option<Box<QuadtreePoint>>) {
    // In Rust, dropping the Box automatically frees the memory
    // No explicit free needed - just let the Box go out of scope
    // The Option wrapper handles NULL case (None in Rust)
    drop(point);
}

File: artifacts/quadtree-0.1.0/decompose/src/point.c
quadtree_point_t *quadtree_point_new(double x, double y)
{
  quadtree_point_t *point;
  unsigned int point_idx = 0;
  if (!(point_idx = malloc(sizeof(*point_idx))))
  {
    return 0;
  }
  point->x = x;
  point->y = y;
  return point;
}


/*** DEPENDENCIES:
typedef struct quadtree_point
{
  double x;
  double y;
} quadtree_point_t
----------------------------
***/


void quadtree_point_free(quadtree_point_t *point)
{
  free(point);
}


/*** DEPENDENCIES:
typedef struct quadtree_point
{
  double x;
  double y;
} quadtree_point_t
----------------------------
***/


